pkg/controller/nstemplateset/nstemplatetier_test.go

```diff
@@ -162,7 +162,7 @@ func TestGetTierTemplate(t *testing.T) {
 		t.Run("host cluster not available", func(t *testing.T) {
 			// given
 			hostCluster := test.NewGetHostCluster(cl, false, apiv1.ConditionFalse)
-			defer resetCache()
+			resetCache()
```

Comment:
why not calling it with `defer`? I want to reset the cache at the end of the test.

Reply:
I thought that for this particular test, we needed to reset the cache before calling `getTierTemplate`?

Reply:
It should not happen that the cache is not reset at the beginning of any test - that's why we always call it using `defer` to make sure that the cache is always cleaned at the end of the test.